eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -259,15 +264,11 @@ object NodeRelayer {
    * should return upstream.
    */
   private def translateError(failures: Seq[PaymentFailure], outgoingNodeId: PublicKey): Option[FailureMessage] = {
-    def tooManyRouteNotFound(failures: Seq[PaymentFailure]): Boolean = {
-      val routeNotFoundCount = failures.collect { case f@LocalFailure(_, RouteNotFound) => f }.length
-      routeNotFoundCount > failures.length / 2
-    }
-
+    val routeNotFound = failures.collectFirst { case f@LocalFailure(_, RouteNotFound) => f }.nonEmpty
     failures match {
       case Nil => None
-      case LocalFailure(_, PaymentError.BalanceTooLow) :: Nil => Some(TemporaryNodeFailure) // we don't have enough outgoing liquidity at the moment
-      case _ if tooManyRouteNotFound(failures) => Some(TrampolineFeeInsufficient) // if we couldn't find routes, it's likely that the fee/cltv was insufficient
+      case LocalFailure(_, BalanceTooLow) :: Nil => Some(TemporaryNodeFailure) // we don't have enough outgoing liquidity at the moment
+      case _ if routeNotFound => Some(TrampolineFeeInsufficient) // if we couldn't find routes, it's likely that the fee/cltv was insufficient
```

Comment:
What's the rationale behind the change in implementation between `tooManyRouteNotFound` and `routeNotFound`?

Reply:
This was a legacy hack :)

The behavior should have always been: if we have a `RouteNotFound` error, that means we couldn't find a route to our destination, so we should ask the sender to retry with a higher fee budget.

But because of the previous MPP split algorithm that looked at channels "blindly", it could generate some `RouteNotFound`s that weren't really `RouteNotFound`s, so we had to relax the condition to "if there are many route not found errors".